File: models/user_model.js
const mongoes = require("mongoose");

const User = new mongoes.Schema({
  name: {
    type: String,
    required: true,
    trim: true,
  },
  email: {
    type: String,
    required: true,
    unique: true,
    trim: true,
  },
  age: {
    type: Number,
    required: true,
  },
});

//crate cluster according to schema
const UserReference = mongoes.model("Users", User);

module.exports = UserReference;

File: lib/models/user_model.dart
class User {
  final String name;
  final String id;
  final String email;
  final int age;

  User(
      {required this.name,
      required this.id,
      required this.email,
      required this.age});

  factory User.fromJson(Map<String, dynamic> json) {
    return User(
        name: json["name"],
        id: json["_id"],
        email: json["email"],
        age: json["age"]);
  }

  Map<String, dynamic> toJson() {
    return {
      "name": name,
      "email": email,
      "age": age,
    };
  }
}
